### ISSUE TITLE
Per-result cursor support (v1beta3)

### ISSUE THREAD

**Issue by blowmage** [GoogleCloudPlatform/gcloud-ruby#711]
Datastore v1beta3 adds a [cursor for every result](https://cloud.google.com/datastore/reference/rest/v1beta3/EntityResult#FIELDS.cursor) returned. We don't currently expose this, and only show the cursor for the last result on the page. The QueryResults object should provide cursors for every result.

```ruby
tasks = datastore.run query
tasks.size #=> 3

# iterate through results and cursors
tasks.each_with_cursor do |task, cursor|
  # ..
end
```

**Comment by timanovsky** [comment]
Is it related to #690 ?

**Comment by blowmage** [comment]
Kinda. I'm working on both.

**Comment by blowmage** [comment]
Resolved in the 0.11.0 release.
